__tests__/app.test.js

```diff
@@ -228,3 +228,94 @@ describe('POST /api/articles/:article_id/comments', () => {
     expect(res.body.message).toBe('bad request');
   });
 });
+
+describe('GET /api/articles queries', () => {
+  test('200: sorted by date descending as default', async () => {
+    const res = await request(app).get('/api/articles').expect(200);
+    expect(res.body).toBeSortedBy('created_at', { coerce: true });
+  });
+  test('200: sorted by any given column in descending order as default', async () => {
+    const res = await request(app)
+      .get('/api/articles?sort_by=article_id')
+      .expect(200);
+    expect(res.body).toBeSortedBy('article_id', {
+      coerce: true,
+      descending: true,
+    });
+    const res2 = await request(app)
+      .get('/api/articles?sort_by=comment_count')
+      .expect(200);
+    expect(res2.body).toBeSortedBy('comment_count', {
+      coerce: true,
+      descending: true,
+    });
+  });
+  test('200: sorted by any given column in choice of descending/ascending order', async () => {
+    const res = await request(app)
+      .get('/api/articles?sort_by=article_id&order=asc')
+      .expect(200);
+    expect(res.body).toBeSortedBy('article_id', {
+      coerce: true,
+      ascending: true,
+    });
+    const res2 = await request(app)
+      .get('/api/articles?sort_by=comment_count&order=asc')
+      .expect(200);
+    expect(res2.body).toBeSortedBy('comment_count', {
+      coerce: true,
+      ascending: true,
+    });
+  });
+  test('200: returns in order of choice sorted by date when given no column', async () => {
+    const res = await request(app).get('/api/articles?order=asc').expect(200);
+    expect(res.body).toBeSortedBy('created_at', {
+      coerce: true,
+      ascending: true,
+    });
+    const res2 = await request(app).get('/api/articles?order=desc').expect(200);
+    expect(res2.body).toBeSortedBy('created_at', {
+      coerce: true,
+      descending: true,
+    });
+  });
+  test('200: returns articles on a given topic', async () => {
+    const res = await request(app).get('/api/articles?topic=mitch').expect(200);
+    res.body.forEach((article) => {
+      expect(article.topic).toBe('mitch');
+    });
+  });
+  test('200: returns an empty array when given a topic which exists but has no articles', async () => {
```

Comment:
Are you sure that this should return a 200 status?

Reply:
My thinking was it isn't a bad request and it isn't 'not found' either technically because the topic exists, so the client can see that the topic exists but there aren't any existing articles atm. Correct me if I'm wrong

Reply:
Remember how we were saying that a 200 status is like finding a box with contents, but that a 204 status is for finding a box with nothing inside of it?